pkg/clients/database/rds.go

```diff
@@ -891,3 +891,42 @@ func areSameElements(a1, a2 []string) bool {
 
 	return true
 }
+
+// DiffTags between spec and current
+func diffTags(spec []v1beta1.Tag, current []rdstypes.Tag) (addTags []rdstypes.Tag, removeTags []string) {
+	currentMap := make(map[string]string, len(current))
+	for _, t := range current {
+		currentMap[pointer.StringValue(t.Key)] = pointer.StringValue(t.Value)
+	}
+
+	specMap := make(map[string]string, len(spec))
+	for _, t := range spec {
+		key := t.Key
+		val := t.Value
+		specMap[key] = t.Value
+
+		if currentVal, exists := currentMap[key]; exists {
+			if currentVal != val {
+				removeTags = append(removeTags, t.Key)
```

Comment:
I don't think this is necessary, since `AddTagsToResource` will overwrite the value of the existing tags?

Reply:
Yes you're right. For the case "same key, changed value" `AddTagsToResource` is enough, no `RemoveTagsFromResource` needed. I will remove the line here.

For the resources in the `rds` APIGroup this could/should be checked too in the future:
https://github.com/crossplane-contrib/provider-aws/blob/4124fb7ccb72938dcfe5ada372d910f81884a270/pkg/controller/rds/utils/tags.go#L107